Rewrite and extend `Client/Beatmap editor/Timing` (#9034)
ppy/osu-wiki: wiki/Client/Beatmap_editor/Timing/en.md

---
tags:
  - red line
  - red offset
  - red timing point
  - uninherited offset
  - green line
  - green offset
  - green timing point
  - inherited offset
  - timing setup
  - copy timing
  - paste timing
---

# Timing

*For a timing setup guide, see: [How to time songs](/wiki/Guides/How_to_Time_Songs)*\
*See also: [Beatmapping/Timing](/wiki/Beatmapping/Timing)*

> **Timing** is the [beatmap editor](/wiki/Client/Beatmap_editor)'s tab for changing and setting up timing of a [beatmap](/wiki/Beatmap), which is essential for correct song representation. It contains related settings and helpers, and also a [dedicated window](#timing-setup-panel) for working with multiple [timing points](#timing-points), both for musical structure description and beatmap design purposes.

Reviewer: strictly speaking, there is the timing tab and the concept known as timing, im not sure how practical it is to have both pages have the same name. whilst it is accurate to the name inside the editor, maybe something more easy to understand like the old "timing tab" or otherwise would be better?

> **Timing** is the [beatmap editor](/wiki/Client/Beatmap_editor)'s tab for changing and setting up timing of a [beatmap](/wiki/Beatmap), which is essential for correct song representation. It contains related settings and helpers, and also a [dedicated window](#timing-setup-panel) for working with multiple [timing points](#timing-points), both for musical structure description and beatmap design purposes.

Reviewer: ```suggestion
**Timing** is the [beatmap editor](/wiki/Client/Beatmap_editor)'s tab for changing and setting up the timing of a [beatmap](/wiki/Beatmap), which is essential for correct song representation. It contains timing related settings and tools, and also a [dedicated window](#timing-setup-panel) for working with multiple [timing points](#timing-points), both for musical structure representation and beatmap design purposes.
```

helpers -> tools seems like a better description here, nothing in the timing tab does things for you and helper seems rly humanising?

Reviewer 2: "helper" doesn't mean anything to me (i've never heard this term used to explain a UI before, if that's what it's trying to do).

## Timing points

*See also: [Offset](/wiki/Offset)*

> In [mapping](/wiki/Beatmapping), a *timing point*, colloquially called an *offset*, is a way to apply common settings, such as [timing](/wiki/Beatmapping/Timing), [slider velocity](/wiki/Gameplay/Hit_object/Slider/Slider_velocity) multiplier, or [hitsounds](/wiki/Beatmapping/Hitsound)' volume, to a whole section of the beatmap. In osu!, there are two types of timing points.

Reviewer: ```suggestion
In [mapping](/wiki/Beatmapping), a *timing point*, colloquially called an *offset*, is a way to apply common settings, such as [timing](/wiki/Beatmapping/Timing), [slider velocity](/wiki/Gameplay/Hit_object/Slider/Slider_velocity) multipliers, or [hitsounds](/wiki/Beatmapping/Hitsound) and their respective volumes, to a specific section of a beatmap. In osu!, there are two types of timing points.
```

### Uninherited timing point

::: Infobox
![](img/uninherited-offsets.png "Several uninherited timing points located in the timing setup panel")
:::

> An **uninherited** timing point has its own timing settings. Several such points are used to describe timing changes in the song, such as [tempo](/wiki/Music_theory/Tempo) or beat irregularities, or different size of the [time signature](/wiki/Music_theory/Time_signature). Adding a new uninherited timing point resets the metronome, and makes it count beats from the timing point's offset, specified in milliseconds.

Reviewer: ```suggestion
An **uninherited** timing point has its own timing settings. Several of such points are used to represent timing changes in the song, such as [tempo](/wiki/Music_theory/Tempo), beat irregularities, omitting a barline in [osu!taiko](/wiki/Game_mode/osu!taiko) or [osu!mania](/wiki/Game_mode/osu!mania), or a different size of [time signature](/wiki/Music_theory/Time_signature). Adding a new uninherited timing point resets the metronome, making it count beats from the timing point's offset, specified in milliseconds.
```

Reviewer 3: fwiw we always treat bar line related things as "timing", even tho it might not make a lot of sense at first glance

> Uninherited timing points have red colour on the bottom timeline and in the [timing setup panel](#timing-setup-panel). Due to this, they are often called "red offsets", or "red lines".

Reviewer: ```suggestion
Uninherited timing points have red colour on the bottom timeline and in the [timing setup panel](#timing-setup-panel). Due to this, they are often called "red offsets" or "red lines".
```

### Inherited timing point

::: Infobox
![](img/inherited-offsets.png "Several inherited timing points with different slider velocity multiplier, volume, and kiai settings")
:::

> An **inherited** offset uses currently active timing settings, *inheriting* them from the previous timing point. Inherited timing points are used for the following purposes:

Reviewer: ```suggestion
An **inherited** offset uses the currently active timing settings, *inheriting* them from the previous uninherited timing point. Inherited timing points are used for the following purposes:
```

> - Changing [slider velocity](/wiki/Gameplay/Hit_object/Slider/Slider_velocity) on a certain section of the beatmap

Reviewer: ```suggestion
- Changing [slider velocity](/wiki/Gameplay/Hit_object/Slider/Slider_velocity)
```
none of the other use cases mention a specific section even though they all can do that, might be best to remove?

The timing tab can be opened by pressing `F3` and provides quick access to the following settings:

| Name | Meaning |
| :-- | :-- |
| `BPM` | [Tempo](/wiki/Music_theory/Tempo) of the current timing point, measured in beats per minute (BPM). |
| `Offset` | [Offset](/wiki/Offset#mapping) of the current timing point, measured in milliseconds. |
| `Move already placed notes when changing the offset/BPM` | Keep hit objects attached to the [timeline](/wiki/Client/Beatmap_editor/Timelines) ticks during timing changes |
| `Slider Velocity` | Base [slider velocity](/wiki/Gameplay/Hit_object/Slider/Slider_velocity) of the whole beatmap |
| `Slider Tick Rate` | How many [slider ticks](/wiki/Gameplay/Hit_object/Slider/Slider_tick) will be in a musical [beat](/wiki/Music_theory/Beat) |

To quickly make larger or smaller changes, use the below modifiers:

|  | Tempo | Offset | Slider velocity |
| :-- | :--: | :--: | :--: |
| `Ctrl` + click | 0.25 BPM | 1 ms | 1 [osu! pixel](/wiki/Client/Beatmap_editor/osu!_pixel) |
| Default | 1 BPM | 2 ms | 10 osu! pixels |
| `Shift` + click | 5 BPM | 10 ms | - |

### Metronome

![](img/metronome.png "The osu! metronome from the timing tab")

> The metronome, which is located in the upper right corner of the timing tab, helps quickly find approximate timing settings. The initial BPM and offset values of the song can be found by repeatedly tapping `T`, or clicking the `Tap Here!` button, while following the song's rhythm.

Reviewer: ```suggestion
The metronome, which is located in the upper right corner of the timing tab, helps to quickly find approximate timing settings. The initial BPM and offset values of the song can be found by repeatedly tapping `T`, or clicking the `Tap Here!` button whilst following the song's rhythm.
```

> The visual part of the metronome is tailored to the most popular [time signature](/wiki/Music_theory/Time_signature), 4/4. The first section denotes the [downbeat](/wiki/Music_theory/Downbeat) and flashes green, while remaining sections flash white to every beat of the song's [measures](/wiki/Music_theory/Measure). Uncommon signatures, such as 7/4, add extra flashes, but the size of the metronome doesn't change.

Reviewer: ```suggestion
The visual part of the metronome is tailored to the most common [time signature](/wiki/Music_theory/Time_signature), 4/4. The first section denotes the [downbeat](/wiki/Music_theory/Downbeat) and flashes green, while remaining sections flash white to every beat of the song's [measures](/wiki/Music_theory/Measure). Uncommon signatures, such as 7/4, add extra flashes, but the size of the metronome doesn't change.
```

Author: yeah

## Timing setup panel

![Screenshot of the timing setup panel](/wiki/shared/timing/TimingSetup.png)

> The timing setup panel, also called `Timing and Control Points`, is a pop-up window that can be opened by pressing `F6`. It contains several tabs for working with different options of the timing points, such as timing, [hitsounds](/wiki/Beatmapping/Hitsound) volume and [sampleset](/wiki/Beatmapping/Sampleset), or special effects.

Reviewer: ```suggestion
The timing setup panel, also called `Timing and Control Points`, is a pop-up window that can be opened by pressing `F6`. It contains several tabs for working with different options of the timing points, such as timing, [hitsounds](/wiki/Beatmapping/Hitsound), volumes, [samplesets](/wiki/Beatmapping/Sampleset), or special effects.
```

Author: your edit makes it sound like it's possible to add a standalone sample to a hit object from the timing panel directly, is that true?

Reviewer: yes, if one specific green line's purpose is to effect one object then that is possible, its easy to see in keysounded mapsets
